pkg/gardenlet/bootstrap/certificate/certificate.go

```diff
@@ -88,23 +88,25 @@ func requestCertificate(ctx context.Context, logger logrus.FieldLogger, certific
 
 	logger.Info("Creating certificate signing request...")
 
-	// LegacyUnkownSignerName based on https://github.com/kubernetes/kubernetes/blob/db4ca87d9d872b3c31df58860bac996c70df6b5b/pkg/apis/certificates/v1beta1/defaults.go#L52.
-	req, err := csr.RequestCertificate(certificateClient, csrData, name, certificatesv1beta1.LegacyUnknownSignerName, usages, privateKey)
+	// TODO (timebertt): figure out correct signerName for gardenlet CSRs
+	// kubernetes.io/legacy-unknown is disallowed when creating new CSR objects in certificates/v1,
+	// ref https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1513-certificate-signing-request#api-changes-between-v1beta1-and-v1
+	reqName, reqUID, err := csr.RequestCertificate(client, csrData, name, certificatesv1.KubeAPIServerClientSignerName, usages, privateKey)
```

Comment:
As discussed, `certificatesv1.KubeAPIServerClientSignerName` should be fine when reading  https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers and comparing the usage field [here](https://github.com/gardener/gardener/blob/221e5d764c43e3ebde6c096d7dc2755521454d13/pkg/controllermanager/controller/certificatesigningrequest/csr_autoapprove_control.go#L195). GCM should also be allowed to approve the request since it has the `cluster-admin` role.

Reply:
Thanks for checking the details. I will remove the `TODO` once I've successfully tested the gardenlet bootstrapping cert rotation :)